x/monitoringc/keeper/verified_client_id.go

```diff
@@ -14,6 +14,12 @@ func (k Keeper) SetVerifiedClientID(ctx sdk.Context, verifiedClientID types.Veri
 	store.Set(types.VerifiedClientIDKey(verifiedClientID.LaunchID), b)
 }
 
+// ClearVerifiedClientIDs removes a set of verifiedClientID in the store from its launch ID
+func (k Keeper) ClearVerifiedClientIDs(ctx sdk.Context, launchID uint64) {
```

Comment:
We must also remove all entries of `LaunchIDFromVerifiedClientID` for each of the client IDs in the array. This is the actual entry read when doing the handshake, hence the error I mentioned above